### ISSUE TITLE
Support Sitemap Index Input

### ISSUE THREAD

**Issue by localnerve** [localnerve/html-snapshots#32]
Extend this library to support a larger input set from a [sitemap index](https://support.google.com/webmasters/answer/75712?hl=en).
Some implementation suggestions:
1. Add a new input generator "sitemap-index".
2. Refactor "sitemap" input generator's functions to a module for use by both sitemap and sitemap-index

**Comment by chrisheijnen** [comment]
Is it possible to get support for this?

**Comment by localnerve** [comment]
I'm available for part-time side-contract work (I'm currently occupied FT until EOY). I'm also available to review others' PR contributions at night (Eastern).

**Comment by codekraft-studio** [comment]
@localnerve 

Hi Alex, i've done something for the sitemap-index input generator.
Can you check the [code](https://github.com/codekraft-studio/html-snapshots/commit/04da436125823acc9709233f99aa97dd1b60ed0c) before I make the pull req?
I did this: 
* moved the **getSitemapUrl**, **getSitemapFile** and **convert** functions in common
* passed the parse function as argument (**next**) of the convert function so sitemap and sitemap-index share the same common funtions but with a different parser
* inside the sitemap-index parser, after parsing the sitemap it **calls the htmlSnapshot function for every sitemap**

I'm unable to add the tests for sitemap-index to the existing ones in project, im not very good with testing.

But i've tested with some sitemap-indexes and you can test with this: https://core.codekraft.it/public/wordpress/sitemap_index.xml

Example script:
```javascript
htmlSnapshots.run({
  input: "sitemap-index",
  source: "https://core.codekraft.it/public/wordpress/sitemap_index.xml",
  outputDir: path.join(__dirname, "./tmp"),
  outputDirClean: true
}, function(err, completed) {  

  console.log("completed snapshots:");
  console.log(util.inspect(completed));

  // throw if there was an error
  assert.ifError(err);
});
```

Seems to work, but i'm pretty sure there is a better way to do it.

**Comment by localnerve** [comment]
@codekraft-studio Thanks! Just looked briefly today, looks cool so far. I'll give it a more thorough look-over in about 4 hours.

**Comment by localnerve** [comment]
@codekraft-studio I can't reference the test index file https://core.codekraft.it/public/wordpress/sitemap_index.xml. Online?

**Comment by codekraft-studio** [comment]
@localnerve I apologize, I forgot that you would have tested and I turned off the server before I went to sleep!

**Comment by localnerve** [comment]
@codekraft-studio 
html-snapshots#rc-0.14.0-4 contains the beginning of an implementation of sitemap-index.
I've still got some tests to write that cover `lastmod` behavior and integration.

Can you offer another live sitemap-index to test against?
https://core.codekraft.it/public/wordpress/sitemap_index.xml has a bad cert:
```
$ curl -i 'https://core.codekraft.it/public/wordpress/sitemap_index.xml'
curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```

**Comment by localnerve** [comment]
@codekraft-studio your sitemap-index test site is working now. I fixed some bugs using your very helpful site. The [sitemap-index example](https://github.com/localnerve/html-snapshots/tree/rc-0.14.0-4/examples/sitemap-index) is serving as a working integration test for now.
I still need to write the actual tests, but I feel like this is probably working.

**Comment by codekraft-studio** [comment]
@localnerve awesome! I wasn't able to do it, i'm not so practice with node right now, but I looked your code and is awesome i'm learning a lot, but in the example you should put this: _https://yoast.com/sitemap_index.xml_
instead of mine, because may be turned off for long periods sometimes and the demo can fail.
I use the yoast sitemap generator for that development site so I assume the yoast original sitemap can be more reliable :)

**Comment by localnerve** [comment]
I'll remove all requests to your site soon (definitely before the release).

Yoast has a great test sitemap, but probably bad for an example because it is so large.
Google also has a great test sitemap, but it also results in a large output.
It would be great to find one that can complete with ~25? files or so.

FWIW, I'm still testing this implementation (most of the work is in testing), I'm still finding small issues (around failure modes). More to come.

**Comment by codekraft-studio** [comment]
I can imagine, you can still use my site for testing no worries, but sometimes you can find it down.
I will try to find on google some example sitemap to use it for the cause.

**Comment by localnerve** [comment]
I found tech.jet.com. in the example now. A reasonable size for now.

**Comment by localnerve** [comment]
Published v0.14.0.
